manifests/agent/replication.pp

```diff
@@ -53,7 +53,7 @@
   Optional[Boolean] $trigger_onoff_time    = undef,
   String $username
 ) {
-  if $name =~ /^[A-Za-z0-9\-_]+$/ {
+  if $name =~ /^[A-Za-z0-9_-]*$/ {
```

Comment:
Met een * betekent 0 of meer occurrences. Dus de + was toch juist? Ik denk eerder dat de backslash het issue was.

Reply:
https://www.puppet.com/docs/puppet/7/lang_data_regexp.html#lang_data_regexp-regex-syntax

Als ik zo wat zaken probeer, dan leek het **voor** de change juist:

```
at 10:34:57 ❯ ruby -e '$name = "a"; puts($name =~ /^[A-Za-z0-9-_]+$/)'
-e:1: warning: character class has '-' without escape: /^[A-Za-z0-9-_]+$/
0

~
at 10:35:08 ❯ ruby -e '$name = "a"; puts($name =~ /^[A-Za-z0-9\-_]+$/)'
0

~
at 10:35:21 ❯ ruby -e '$name = "a"; puts($name =~ /^[A-Za-z0-9\-_]*$/)'
0

~
at 10:35:26 ❯ ruby -e '$name = ""; puts($name =~ /^[A-Za-z0-9\-_]+$/)'


~
at 10:35:35 ❯ ruby -e '$name = ""; puts($name =~ /^[A-Za-z0-9\-_]*$/)'
0

~
at 10:35:41 ❯ ruby -e '$name = ""; puts($name =~ /^[A-Za-z0-9\-_]+$/)'


~
at 10:35:46 ❯ ruby -e '$name = "publish"; puts($name =~ /^[A-Za-z0-9\-_]+$/)'
0
```

Reply:
Inderdaad, PR updated, `=~` moest `!~` zijn in deze logica.